Scripts/BuildPhases/LintAppLocalizedStringsUsage.swift

```diff
@@ -0,0 +1,324 @@
+    /// - note: Does not represent other types of Build Phases that could exist in the project, only "Compile Sources" one
+    struct PBXSourcesBuildPhase: PBXObject {
+        private let isa: ISA<Self>
+
+        let files: [ObjectUUID]
+    }
+
+    /// One of the many `PBXObject` types encountered in the `.pbxproj` file format.
+    /// Represents a single build file in a `PBXSourcesBuildPhase` build phase.
+    struct PBXBuildFile: PBXObject {
+        private let isa: ISA<Self>
+
+        let fileRef: ObjectUUID
+    }
+
+    /// This type is used to indicate what a file reference in the project is actually relative to
+    enum SourceTree: String, Decodable {
+        case absolute = "<absolute>"
+        case group = "<group>"
+        case projectRoot = "SOURCE_ROOT"
+        case buildProductsDir = "BUILT_PRODUCTS_DIR"
+        case devDir = "DEVELOPER_DIR"
+        case sdkDir = "SDKROOT"
+    }
+
+    /// One of the many `PBXObject` types encountered in the `.pbxproj` file format.
+    /// Represents a reference to a file contained in the project tree.
+    struct PBXFileReference: PBXReference {
+        private let isa: ISA<Self>
+
+        let name: String?
+        let path: String?
+        let sourceTree: SourceTree
+    }
+
+    /// One of the many `PBXObject` types encountered in the `.pbxproj` file format.
+    /// Represents a group (aka "folder") contained in the project tree.
+    struct PBXGroup: PBXReference {
+        enum ISA: String, Decodable { case PBXGroup, XCVersionGroup, PBXVariantGroup }
+        private let isa: ISA
+
+        let name: String?
+        let path: String?
+        let sourceTree: SourceTree
+        let children: [ObjectUUID]
+    }
+
+    /// Fallback type for any unknown `PBXObject` type.
+    struct UnknownPBXObject: PBXObject {
+        let isa: String
+    }
+
+    /// Wrapper helper to decode any `PBXObject` based on the value of their `isa` field
+    @propertyWrapper
+    struct PBXObjectWrapper: Decodable, CustomDebugStringConvertible {
+        let wrappedValue: PBXObject
+        static let knownTypes: [PBXObject.Type] = [
+            PBXProject.self,
+            PBXGroup.self,
+            PBXFileReference.self,
+            PBXNativeTarget.self,
+            PBXSourcesBuildPhase.self,
+            PBXBuildFile.self
+        ]
+
+        init(from decoder: Decoder) throws {
+            // Try to decode each known types in turn, until we find one that succeeds decoding — by having the expected `isa` field value.
+            for objectType in Self.knownTypes {
+                if let object = try? objectType.init(from: decoder) as PBXObject {
+                    self.wrappedValue = object
+                    return
+                }
+            }
+            self.wrappedValue = try UnknownPBXObject(from: decoder) as PBXObject // Fallback
+        }
+        var debugDescription: String { String(describing: wrappedValue) }
+    }
```

Comment:
Inspired by https://forums.swift.org/t/decoding-to-different-models-based-on-json-key/19288

I tried other approaches to solve the "decode the right subtype based on a field/key (i.e. `isa`), but this ended up being the simpler and less convoluted one, and also the most performant (especially compared to the other I tried of decoding the dictionary twice — first one to just extract the `isa` and determine the proper type, and another one to reparse the entire dictionary to instantiate said type — which slowed down considerably the parsing, at least when I tested it in my Swift Playground)

Reply:
Actually I couldn't help myself but play with Swift over the weekend to try other ideas, and found one that I think might look neater. I might update the code to use it at some point, either in this PR or in a next one.